pkg/export/transform.go

```diff
@@ -432,3 +463,78 @@ Loop:
 	// Batch ended without completing a further distribution
 	return nil, 0, samples[consumed:], nil
 }
+
+func buildExemplars(exemplars []record.RefExemplar) []*distribution_pb.Distribution_Exemplar {
+	// the exemplars field of a distribution value field must be in increasing order of value
+	// (https://cloud.google.com/monitoring/api/ref_v3/rpc/google.api#distribution) -- let's sort them
+	sort.Slice(exemplars, func(i, j int) bool {
+		return exemplars[i].V < exemplars[j].V
+	})
+	var result []*distribution_pb.Distribution_Exemplar
+	for _, promExemplar := range exemplars {
+		attachments := buildExemplarLabels(promExemplar.Labels)
+		result = append(result, &distribution_pb.Distribution_Exemplar{
+			Value:       promExemplar.V,
+			Timestamp:   getTimestamp(promExemplar.T),
+			Attachments: attachments,
+		})
+	}
+	return result
+}
+
+// buildExemplarLabels transforms the prometheus LabelSet into a GCM exemplar attachment.
+// If the following three fields are present in the LabelSet, then we will build a SpanContext:
```

Comment:
Maybe highlight here?
```
Note that the project_id needs to be the project_id where the span was written. This may not necessarily be the same project_id where the metric is written
 ```